nuget-tools.cake

```diff
@@ -0,0 +1,219 @@
+}
+
+async Task<SourcePackageDependencyInfo> GetDependency(DependencyInfoResource dependencyResource, NugetDependency dependency)
+{
+    Information($"GetDependency {dependency.Name} version {dependency.Version.ToString()}");
+    return await dependencyResource.ResolvePackage(new PackageIdentity(dependency.Name, dependency.Version), dependency.Framework, new SourceCacheContext(), NullLogger.Instance, CancellationToken.None); 
+}
+
+DependencyInfoResource GetDefaultDependencyResource() 
+{
+    var sourceRepository = new SourceRepository(new PackageSource(NuGetConstants.V3FeedUrl), Repository.Provider.GetCoreV3());
+    return sourceRepository.GetResource<DependencyInfoResource>(CancellationToken.None);
+}
+
+async Task ProcessDependency(NugetDependency dependency, string destination) 
+{
+    var package = await GetDependency(GetDefaultDependencyResource(), dependency);
+    var uri = package.DownloadUri.ToString();
+    Information($"Downloading {package.Id} from {uri}");
+    DownloadFile(uri, ExternalsFolder + package.Id + ".nupkg");
+    Information($"Extract NuGet package: {dependency.Name}");
+    var path = ExternalsFolder + dependency.Name + ".nupkg";
+    var tempPackageFolder = ExternalsFolder + dependency.Name;
+    PackageExtractor.Extract(path, tempPackageFolder);
+    ProcessPackageDlls(dependency, tempPackageFolder, destination);
+}
+
+string GetNuGetPackage(string packageId, string packageVersion)
+{
+    var nugetUser = EnvironmentVariable("NUGET_USER");
+    var nugetPassword = Argument("NuGetPassword", EnvironmentVariable("NUGET_PASSWORD"));
+    var nugetFeedId = Argument("NuGetFeedId", EnvironmentVariable("NUGET_FEED_ID"));
+    packageId = packageId.ToLower();
+    var filename = packageId + "." + packageVersion +  ".nupkg";
+    var url = $"{PrivateNugetFeedUrl}{nugetFeedId}/nuget/v3/flat2/{packageId}/{packageVersion}/{filename}";
+
+    // Get the NuGet package
+    HttpWebRequest request = (HttpWebRequest)WebRequest.Create(url);
+    request.Headers["X-NuGet-ApiKey"] = nugetPassword;
+    request.Credentials = new NetworkCredential(nugetUser, nugetPassword);
+    HttpWebResponse response = (HttpWebResponse)request.GetResponse();
+    var responseString = String.Empty;
+    using (var fstream = new FileStream(filename, FileMode.Create, FileAccess.ReadWrite))
+    {
+        response.GetResponseStream().CopyTo(fstream);
+    }
+    return filename;
+}
+
+FilePathCollection ResolveDllFiles(string tempContentPath, NuGetFramework frameworkName) 
+{
+    var destinationPath = $"{tempContentPath}/lib/{frameworkName.GetShortFolderName()}/*.dll";
+    var files = GetFiles(destinationPath);
+    if (files.Any()) 
+    {
+        return files;
+    }
+    Warning($"Haven't found anything under {tempContentPath} - make sure it's expected.");
+    return null;
+}
+
+void MoveNativeBinaries(string tempContentPath, string destination)
+{
+    var runtimesPath = tempContentPath + "/runtimes";
+    if (DirectoryExists(runtimesPath)) 
```

Comment:
Let's revert the condition and return. This will reduce indent and improve readability